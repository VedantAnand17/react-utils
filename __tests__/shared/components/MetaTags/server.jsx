/**
 * Tests for the server-side MetaTags integration.
 */

import supertest from 'supertest';
import { Helmet } from 'react-helmet';

import serverFactory from 'server/server';

import Application, { MODES } from './__assets__/testcase/app';
import Application2, { MODES as MODES2 } from './__assets__/testcase/app2';

// Without this setting the JSDom presence in test environment tricks
// tricks react-helmet to think it is executed client-side.
Helmet.canUseDOM = false;

const WEBPACK_CONFIG = {
  context: `${__dirname}/__assets__/testcase`,
  output: {
    path: '/path/to/assets',
    publicPath: '/test/public/path',
  },
};

async function baseTest(Component) {
  const server = supertest(
    await serverFactory(WEBPACK_CONFIG, {
      Application: Component,
      beforeRender: () => ({
        initialState: { domain: 'https://sample.domain' },
      }),
      logger: {
        ...console,
        info: () => null,
      },
    }),
  );
  await server.get('/').expect(200)
    .expect((res) => expect(res.text).toMatchSnapshot());
}

test('Basic tags, no override', () => baseTest(
  () => <Application mode={MODES.BASIC_NO_OVERRIDE} />,
));
test('Basic tags, with override', () => baseTest(
  () => <Application mode={MODES.BASIC_WITH_OVERRIDE} />,
));
test('All tags, with override', () => baseTest(
  () => <Application mode={MODES.ALL_TAGS_WITH_OVERRIDE} />,
));

describe('Using <MetaTags> children feature', () => {
  test('Basic tags, no override', () => baseTest(
    () => <Application2 mode={MODES2.BASIC_NO_OVERRIDE} />,
  ));
  test('Basic tags, with override', () => baseTest(
    () => <Application2 mode={MODES2.BASIC_WITH_OVERRIDE} />,
  ));
});
