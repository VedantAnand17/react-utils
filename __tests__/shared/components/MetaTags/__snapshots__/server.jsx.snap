// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All tags, with override 1`] = `
"<!DOCTYPE html>
        <html lang=\\"en\\">
          <head>
            
            <title data-react-helmet=\\"true\\">Title from Component</title>
            <meta data-react-helmet=\\"true\\" name=\\"description\\" content=\\"Component description\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:card\\" content=\\"summary_large_image\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:title\\" content=\\"SOCIAL_TITLE\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:description\\" content=\\"SOCIAL_DESCRIPTION\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:image\\" content=\\"https://sample.domain/THUMBNAIL_PATH\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:site\\" content=\\"@SITE_NAME\\"/><meta data-react-helmet=\\"true\\" name=\\"og:title\\" content=\\"SOCIAL_TITLE\\"/><meta data-react-helmet=\\"true\\" name=\\"og:image\\" content=\\"https://sample.domain/THUMBNAIL_PATH\\"/><meta data-react-helmet=\\"true\\" name=\\"og:image:alt\\" content=\\"SOCIAL_TITLE\\"/><meta data-react-helmet=\\"true\\" name=\\"og:description\\" content=\\"SOCIAL_DESCRIPTION\\"/><meta data-react-helmet=\\"true\\" name=\\"og:sitename\\" content=\\"SITE_NAME\\"/><meta data-react-helmet=\\"true\\" name=\\"og:url\\" content=\\"PAGE_URL\\"/>
            <meta name=\\"theme-color\\" content=\\"#FFFFFF\\">
            
            
            
            <meta charset=\\"utf-8\\">
            <meta
              content=\\"width=device-width,initial-scale=1.0\\"
              name=\\"viewport\\"
            >
          </head>
          <body>
            
            <div id=\\"react-view\\"><div><p>Hello World!</p><div><p>A dummy internal component.</p></div></div></div>
            <script
              id=\\"inj\\"
              type=\\"application/javascript\\"
              nonce=\\"00000000-0000-0000-0000-000000000003\\"
            >
              window.INJ=\\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghgc5WRH69eBKHiBZQmMDPrDIZjS0mo8ysGCjayxGzD6WmL8eLQdJGGbkbV9YPqFyAc1uVFm6WcpMG4mD/Npa7cqgK6UVPOjCVTlFbZPBYhN1gUlx9R+UmuUKa6MNZMcJG\\"
            </script>
            
            
          </body>
        </html>"
`;

exports[`Basic tags, no override 1`] = `
"<!DOCTYPE html>
        <html lang=\\"en\\">
          <head>
            
            <title data-react-helmet=\\"true\\">Application Title</title>
            <meta data-react-helmet=\\"true\\" name=\\"description\\" content=\\"Application Description\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:card\\" content=\\"summary_large_image\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:title\\" content=\\"Application Title\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:description\\" content=\\"Application Description\\"/><meta data-react-helmet=\\"true\\" name=\\"og:title\\" content=\\"Application Title\\"/><meta data-react-helmet=\\"true\\" name=\\"og:description\\" content=\\"Application Description\\"/>
            <meta name=\\"theme-color\\" content=\\"#FFFFFF\\">
            
            
            
            <meta charset=\\"utf-8\\">
            <meta
              content=\\"width=device-width,initial-scale=1.0\\"
              name=\\"viewport\\"
            >
          </head>
          <body>
            
            <div id=\\"react-view\\"><div><p>Hello World!</p></div></div>
            <script
              id=\\"inj\\"
              type=\\"application/javascript\\"
              nonce=\\"00000000-0000-0000-0000-000000000001\\"
            >
              window.INJ=\\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghgc5WRH69eBKHiBZQmMDPrDIZjS0mo8ysGCjayxGzD6WmL8eLQdJGGbkbV9YPqFyAc1uVFm6WcpMG4mD/Npa7cqgK6UVPOjCVTlFbZPBYhN1gUlx9R+UmuUKa6MNZMcJG\\"
            </script>
            
            
          </body>
        </html>"
`;

exports[`Basic tags, with override 1`] = `
"<!DOCTYPE html>
        <html lang=\\"en\\">
          <head>
            
            <title data-react-helmet=\\"true\\">Title from Component</title>
            <meta data-react-helmet=\\"true\\" name=\\"description\\" content=\\"Component description\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:card\\" content=\\"summary_large_image\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:title\\" content=\\"Title from Component\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:description\\" content=\\"Component description\\"/><meta data-react-helmet=\\"true\\" name=\\"og:title\\" content=\\"Title from Component\\"/><meta data-react-helmet=\\"true\\" name=\\"og:description\\" content=\\"Component description\\"/>
            <meta name=\\"theme-color\\" content=\\"#FFFFFF\\">
            
            
            
            <meta charset=\\"utf-8\\">
            <meta
              content=\\"width=device-width,initial-scale=1.0\\"
              name=\\"viewport\\"
            >
          </head>
          <body>
            
            <div id=\\"react-view\\"><div><p>Hello World!</p><div><p>A dummy internal component.</p></div></div></div>
            <script
              id=\\"inj\\"
              type=\\"application/javascript\\"
              nonce=\\"00000000-0000-0000-0000-000000000002\\"
            >
              window.INJ=\\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghgc5WRH69eBKHiBZQmMDPrDIZjS0mo8ysGCjayxGzD6WmL8eLQdJGGbkbV9YPqFyAc1uVFm6WcpMG4mD/Npa7cqgK6UVPOjCVTlFbZPBYhN1gUlx9R+UmuUKa6MNZMcJG\\"
            </script>
            
            
          </body>
        </html>"
`;

exports[`Using <MetaTags> children feature Basic tags, no override 1`] = `
"<!DOCTYPE html>
        <html lang=\\"en\\">
          <head>
            
            <title data-react-helmet=\\"true\\">Application Title</title>
            <meta data-react-helmet=\\"true\\" name=\\"description\\" content=\\"Application Description\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:card\\" content=\\"summary_large_image\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:title\\" content=\\"Application Title\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:description\\" content=\\"Application Description\\"/><meta data-react-helmet=\\"true\\" name=\\"og:title\\" content=\\"Application Title\\"/><meta data-react-helmet=\\"true\\" name=\\"og:description\\" content=\\"Application Description\\"/>
            <meta name=\\"theme-color\\" content=\\"#FFFFFF\\">
            
            
            
            <meta charset=\\"utf-8\\">
            <meta
              content=\\"width=device-width,initial-scale=1.0\\"
              name=\\"viewport\\"
            >
          </head>
          <body>
            
            <div id=\\"react-view\\"><div><p>Hello World!</p></div></div>
            <script
              id=\\"inj\\"
              type=\\"application/javascript\\"
              nonce=\\"00000000-0000-0000-0000-000000000004\\"
            >
              window.INJ=\\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghgc5WRH69eBKHiBZQmMDPrDIZjS0mo8ysGCjayxGzD6WmL8eLQdJGGbkbV9YPqFyAc1uVFm6WcpMG4mD/Npa7cqgK6UVPOjCVTlFbZPBYhN1gUlx9R+UmuUKa6MNZMcJG\\"
            </script>
            
            
          </body>
        </html>"
`;

exports[`Using <MetaTags> children feature Basic tags, with override 1`] = `
"<!DOCTYPE html>
        <html lang=\\"en\\">
          <head>
            
            <title data-react-helmet=\\"true\\">Title from Component - Application Title</title>
            <meta data-react-helmet=\\"true\\" name=\\"description\\" content=\\"Component description - Application Description\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:card\\" content=\\"summary_large_image\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:title\\" content=\\"Title from Component - Application Title\\"/><meta data-react-helmet=\\"true\\" name=\\"twitter:description\\" content=\\"Component description - Application Description\\"/><meta data-react-helmet=\\"true\\" name=\\"og:title\\" content=\\"Title from Component - Application Title\\"/><meta data-react-helmet=\\"true\\" name=\\"og:description\\" content=\\"Component description - Application Description\\"/>
            <meta name=\\"theme-color\\" content=\\"#FFFFFF\\">
            
            
            
            <meta charset=\\"utf-8\\">
            <meta
              content=\\"width=device-width,initial-scale=1.0\\"
              name=\\"viewport\\"
            >
          </head>
          <body>
            
            <div id=\\"react-view\\"><div><p>Hello World!</p><div><p>A dummy internal component.</p></div></div></div>
            <script
              id=\\"inj\\"
              type=\\"application/javascript\\"
              nonce=\\"00000000-0000-0000-0000-000000000005\\"
            >
              window.INJ=\\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghgc5WRH69eBKHiBZQmMDPrDIZjS0mo8ysGCjayxGzD6WmL8eLQdJGGbkbV9YPqFyAc1uVFm6WcpMG4mD/Npa7cqgK6UVPOjCVTlFbZPBYhN1gUlx9R+UmuUKa6MNZMcJG\\"
            </script>
            
            
          </body>
        </html>"
`;
